src/team.coffee

```diff
@@ -22,196 +22,156 @@
+  Team.robot = robot
 
-  unless config.admin_list?
-    robot.logger.warning 'The HUBOT_TEAM_ADMIN environment variable not set'
-
-  if config.admin_list?
-    admins = config.admin_list.split ','
-  else
-    admins = []
-
-  teamSize = (team_name, msg) ->
-    return false unless teamExists(team_name, msg)
-    robot.brain.data.teams[team_name].length
-
-  teamLabel = (team_name) ->
-    label = team_name unless team_name is config.default_team_label
-    message = if label then "`#{label}` team" else "team"
-    return message
-
-  addUserToTeam = (user, team_name, msg) ->
-    if not team_name
-      robot.brain.data['teams'][config.default_team_label] ||= []
-      team_name = config.default_team_label
-    else
-      return unless teamExists(team_name, msg)
-
-    if user in robot.brain.data.teams[team_name]
-      msg.send "#{user} already in the #{teamLabel(team_name)}"
-    else
-      count = teamSize(team_name, msg)
-      if count > 0
-        countMessage = ", " + count
-        countMessage += " others are in" if count > 1
-        countMessage += " other is in" if count == 1
-
-      robot.brain.data.teams[team_name].push(user)
-
-      message = "#{user} added to the #{teamLabel(team_name)}"
-      message += countMessage if countMessage
-      msg.send message
-
-  addTeam = (team_name, msg) ->
-    if robot.brain.data.teams[team_name]
-      msg.send "#{teamLabel(team_name)} already exists"
-    else
-      robot.brain.data['teams'][team_name] = []
-      msg.send "#{teamLabel(team_name)} created, add some people to it"
-
-  removeUserFromTeam = (user, team_name, msg) ->
-    team_name = config.default_team_label unless team_name
-    return unless teamExists(team_name, msg)
-
-    if user not in robot.brain.data.teams[team_name]
-      msg.send "#{user} already out of the #{teamLabel(team_name)}"
-    else
-      user_index = robot.brain.data.teams[team_name].indexOf(user)
-      robot.brain.data.teams[team_name].splice(user_index, 1)
-      count = teamSize(team_name, msg)
-      countMessage = ", " + count + " remaining" if count > 0
-      message = "#{user} removed from the #{teamLabel(team_name)}"
-      message += countMessage if countMessage
-      msg.send message
-
-  teamExists = (team_name, msg) ->
-    return true if team_name is config.default_team_label
-
-    if robot.brain.data.teams[team_name]
-      true
-    else
-      msg.send "#{teamLabel(team_name)} does not exist"
-      false
+  unless Config.adminList()
+    robot.logger.warning 'HUBOT_TEAM_ADMIN environment variable not set'
 
   ##
   ## hubot create <team_name> team - create team called <team_name>
   ##
   robot.respond /create (\S*) team ?.*/i, (msg) ->
-    team_name = msg.match[1]
-    addTeam(team_name, msg)
+    teamName = msg.match[1]
+    message = if team = Team.create teamName
+                responseMessage.teamCreated(team)
+              else
+                responseMessage.teamAlreadyExists(teamName)
```

Comment:
This type of syntax looks odd, why not simply:

```js
if Team.exists(teamName)
   message = responseMessage.teamAlreadyExists(teamName)
else
   Team.create(teamName)
   message = responseMessage.teamCreated(team)
```

Which makes it easier to follow and doesn't need you check what the `Team.create` implementation is to see that the check for `exists` happens at that point (I would remove it from there)